process/process_linux.go

```diff
@@ -837,10 +837,14 @@ func AllProcesses() (map[int32]*FilledProcess, error) {
 			log.Debugf("Unable to read process command line for %d: %s", pid, err)
 			cmdline = []string{}
 		}
+
+		ctxSwitchesStat := &NumCtxSwitchesStat{}
 		if err := p.fillFromStatus(); err != nil {
 			log.Debugf("Unable to fill from /proc/%d/status: %s", pid, err)
+		} else {
+			ctxSwitchesStat = p.numCtxSwitches
```

Comment:
Any reason not to just do `p.numCtxSwitches = &NumCtxSwitchesStat{}`?

Reply:
That would subtly change the return of 

```
func (p *Process) NumCtxSwitches() (*NumCtxSwitchesStat, error)
```

to return an empty ctx switches struct. It would probably be okay, but this is a slightly smaller change.

Reply:
Oh sorry, I was thinking the change is here:

```
if err := p.fillFromStatus(); err != nil {
	log.Debugf("Unable to fill from /proc/%d/status: %s", pid, err)
        p.numCtxSwitches = &NumCtxSwitchesStat{}
}
```